### PR TITLE
[LA-114] - Switched to a single credentials table with relevant access permissions for LRS API's

### PR DESCRIPTION
**Pull request by sandeep-jay**
Changes include,
- Removal of cookie parsing login in server modules.
- Changes to the LRS data model where separate read/write credential tables were combined into a single table to maintain all the relevant authorizations with permissions flags  for reads/ writes/ anonymization/ datashare features of the LRS
- Updates to the authentication process for the incoming requests in the authorization middleware.
- Callback and error handling improvements for authorization scenarios 
- Updates to all the LRS APIs to check for relevant access permissions to fulfill the API operations.
- Documentation updates.

### CODE REVIEW

File: lib/lrs-core/db.js [hunk 1]
```diff
@@ -238,100 +238,82 @@ var setUpModel = function(sequelize) {
...
-  // will be used if the read credentials allows reading from all
-  // tenants
-  ReadCredential.belongsTo(Tenant, {
+  // Every credential belongs to a tenant
+  Credential.belongsTo(Tenant, {
     'onDelete': 'CASCADE',
     'foreignKey': {
       'name': 'tenant_id',
-      'allowNull': true
+      'allowNull': false
```
Comment (raydavis) [inline lib/lrs-core/db.js:299]: If I'm following the new logic correctly, it looks like we're moving to global-LRS read and write permissions. What is the Credential's Tenant used for at this point?

**Comment by raydavis** [comment]
Apologies, @sandeepmjay ; I think this is just a memory lapse at the end of the day. Am I right that using the Opt-Out table to filter user-centered requests is a known To-Do?

**Comment by raydavis** [comment]
This looks great, @sandeepmjay , and I'm eager to change the Privacy Dashboard code to match. Hope it's OK that I'm taking the opportunity to clear up a couple of lingering questions.

File: lib/lrs-core/db.js (hunk 1 above)
Comment (sandeep-jay) [reply lib/lrs-core/db.js:299]: A tenant can have multiple integrations with data providers/consumers. For example UCB prod might have integrations with canvas live events-prod, gradescope-prod, privacy dashboard etc. in credentials table. Of these some projects might have restricted permissions like read only which are represented by booleans.

**Pull request merged**